cartographer/mapping/map_builder.cc

```diff
@@ -188,8 +189,16 @@ void MapBuilder::LoadMap(io::ProtoStreamReader* const reader) {
     sparse_pose_graph_->FreezeTrajectory(new_trajectory_id);
   }
 
+  // Apply the calculated remapping to constraints in the SparsePoseGraph proto
+  for (auto& constraint_proto : *pose_graph->mutable_constraint()) {
```

Comment:
When the map is frozen deserializing the constraints provides no benefit but comes at a cost regarding performance and memory. Either make it possible to load unfrozen maps and conditionally deserialize constraints in this case. Or first introduce a function for deserializing the state, i.e. not intended to provide a frozen map but to continue SLAMming. What do you think?

Reply:
That is true. In #554 `LoadMap` gets renamed to `LoadState` and gets a boolean parameter whether to freeze or not. Could we delay this optimization until then? Alternatively, we can add the parameter now, skip deserialization if `frozen = true`, and quit if `frozen = false` because it's not ready yet.

Reply:
Ok, so then let's move this block of code over to #554.